pkg/innerring/rpc.go

```diff
@@ -46,8 +46,10 @@ func newClientCache(p *clientCacheParams) *ClientCache {
 	}
 }
 
-func (c *ClientCache) Get(address string, opts ...client.Option) (*client.Client, error) {
-	return c.cache.Get(c.key, address, opts...)
+func (c *ClientCache) Get(address string) (*client.Client, error) {
+	// Because cache is used by `ClientCache` exclusively,
+	// client will always have valid key.
```

Comment:
I think you missed `c.getSG`, `c.GetHeader` and `c.GetRangeHash` functions. They use `c.Get` to get client, and then use it without new `WithKey` option.

Reply:
Fixed.